machine-runner/tests/esm/runner.test.ts

```diff
@@ -423,6 +372,54 @@ describe('machine as async generator', () => {
     expect(r1.machine.isDestroyed()).toBe(true)
   })
 
+  it('should iterate only on state-change and caughtUp', async () => {
+    const { Off, On } = ProtocolSwitch
+
+    const r = new Runner(On, { toggleCount: 0 })
+    const machine = r.machine
+    r.feed([], true)
+
+    let toggleCount = 0
+    let iterationCount = 0
+
+    for await (const state of machine) {
+      iterationCount += 1
+      toggleCount =
+        state.as(On, (x) => x.payload.toggleCount) ||
+        state.as(Off, (x) => x.payload.toggleCount) ||
+        toggleCount
+
+      const whenOn = state.as(On)
+      if (whenOn) {
+        console.log(state.type, state.payload)
+        if (whenOn.payload.toggleCount > 0) {
+          break
+        }
+
+        // spam toggle commands
+
+        // two of these should go to "locked" case
+        const promises = [
+          whenOn.commands?.toggle(),
+          whenOn.commands?.toggle(),
+          whenOn.commands?.toggle(),
+        ]
+        await Promise.all(promises)
+
+        // this one should go to the expired case
```

Comment:
This basically assumes that the next state has been emitted, right? I don’t think this is guaranteed, though.

Reply:
Ah, you are right. Even if the mock Runner supports this, we can't make this assumption
I'll add sleep here